### PR TITLE
Closes #21

### LINKED ISSUE #21: `DataDual` clean up
To do:

- [x] Revise documentation
  - [x] RStudio outline section dividers (use `# function or class name ----`)
  - [x] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [x] Replace `##'` by `#'`
  - [x] Add to `pkgdown.yml`
  - [x] remove file history and the timestamp in comments on top of file
- [x] Add unit tests
  - [x] for classes:
    - [x] test default constructor
    - [x] for reference classes: test methods
  - [x] for functions: as usual
- [x] Refactor as needed
  - [x] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [x] keep externally used names so that we don't break downstream code
  - [x] consider renaming internal variables etc. if needed
  - [x] if a function is too long, break up into pieces
  - [x] for classes: move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
  - [x] in `setClass` function, just use the `slots` instead of outdated `representation`.
  - [x] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #21

### CODE REVIEW

File: R/Data-class.R [hunk 1]
```diff
@@ -141,58 +141,42 @@ Data <- function(x = numeric(),
...
-                             "w must have length nObs")
+# DataDual-constructor ----
 
-                     o$result()
-                 })
-validObject(.DataDual())
+#' @rdname DataDual-class
+#'
+#' @param w (`numeric`)\cr the continuous vector of biomarker values.
+#' @param ... parameters passed to [`Data`].
```
Comment (danielinteractive) [inline R/Data-class.R:169]:
```suggestion
#' @param ... parameters passed to [Data()].
```
(to the constructor)

File: tests/testthat/test-Data-validity.R [hunk 2]
```diff
@@ -82,3 +82,22 @@ test_that("validate_data returns expected error when first xLevel does not match
...
+
+test_that("validate_data_dual passes for valid object", {
+  object <- h_get_data_dual()
+  expect_true(validate_data_dual(object))
+})
+
+test_that("validate_data_dual returns error for biomarker vector of wrong length", {
+  object <- h_get_data_dual()
+
+  # We assign biomarker vector of length different than object@nObs
```
Comment (danielinteractive) [inline tests/testthat/test-Data-validity.R:96]:
```suggestion
  # We assign biomarker vector of length different than object@nObs.
```

**Review by danielinteractive — APPROVED** [review]
thx Wojciech, nice job!

**Comment by danielinteractive** [comment]
@dinakar29 fyi, we here get:

Warning in texi2dvi(file = file, pdf = TRUE, clean = clean, quiet = quiet,  :
  texi2dvi script/program not available, using emulation
Error: processing vignette 'crmPack-jss-paper.Rnw' failed with diagnostics:
unable to run pdflatex on 'crmPack-jss-paper.tex'
LaTeX errors:
! Package xcolor Error: Undefined color `Red'.

See the xcolor package documentation for explanation.
Type  H <return>  for immediate help.

**Comment by dinakar29** [comment]
> @dinakar29 fyi, we here get:
> 
> Warning in texi2dvi(file = file, pdf = TRUE, clean = clean, quiet = quiet, : texi2dvi script/program not available, using emulation Error: processing vignette 'crmPack-jss-paper.Rnw' failed with diagnostics: unable to run pdflatex on 'crmPack-jss-paper.tex' LaTeX errors: ! Package xcolor Error: Undefined color `Red'.
> 
> See the xcolor package documentation for explanation. Type H for immediate help.

Yes, indeed. This is what I'm trying to diagnose and troubleshoot. Will let you know when I have a resolution for the same. It's a tricky one, that's for sure.

**Pull request merged**